src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -106,6 +100,18 @@ CompositionalMultiphaseBase::CompositionalMultiphaseBase( const string & name,
     setApplyDefaultValue( 1 ).
     setDescription( "Flag indicating whether local (cell-wise) chopping of negative compositions is allowed" );
 
+  // Statistics
+
+  this->registerWrapper( viewKeyStruct::averagePressureString(), &m_averagePressure );
+  this->registerWrapper( viewKeyStruct::minimumPressureString(), &m_minimumPressure );
+  this->registerWrapper( viewKeyStruct::maximumPressureString(), &m_maximumPressure );
+  this->registerWrapper( viewKeyStruct::averageTemperatureString(), &m_averageTemperature );
+  this->registerWrapper( viewKeyStruct::minimumTemperatureString(), &m_minimumTemperature );
+  this->registerWrapper( viewKeyStruct::maximumTemperatureString(), &m_maximumTemperature );
+  this->registerWrapper( viewKeyStruct::totalPoreVolumeString(), &m_totalPoreVolume );
+  this->registerWrapper( viewKeyStruct::phasePoreVolumeString(), &m_phasePoreVolume ).
+    setSizedFromParent( 0 );
```

Comment:
You are attaching/registering/giving a name to 8 member to your solver just to display some convergence information. That sounds heavy.

I would preferably build a `struct` with the proper fields
```c++
struct{
  real64 m_averagePressure;
  real64 m_minimumPressure;
  real64 m_maximumPressure;
  real64 m_averageTemperature;
  real64 m_minimumTemperature;
  real64 m_maximumTemperature;
  real64 m_totalPoreVolume;
  array1d< real64 > m_phasePoreVolume;
}
```
and get your `computeReservoirStatistics` build and return it. Then you could display it in your `computeStatistics`.

Reply:
To simplify the code, I removed the `averageTemperature`, `minTemperature`, and `maxTemperature` so there are not fewer quantities to register. Also, I removed all the member variables `m_X` used for statistics (I just register data on the mesh, without using member variables).